src/certifier.cc

```diff
@@ -25,444 +25,212 @@ using namespace certifier::utilities;
-  delete deleted;
-}
 
-int certifier::framework::policy_store::get_num_storage_info() {
-  return num_si_;
-}
-
-const storage_info_message* certifier::framework::policy_store::get_storage_info_by_index(int n) {
-  if (n >= num_si_)
-    return nullptr;
-  return si_[n];
-}
-
-bool certifier::framework::policy_store::add_storage_info(storage_info_message& to_add) {
-  if ((num_si_ + 1) >= max_num_si_)
-    return false;
-  storage_info_message* t = new(storage_info_message);
-  int n = num_si_++;
-  si_[n] = t;
-  si_[n]->CopyFrom(to_add);
-  return true;
-}
-
-void certifier::framework::policy_store::delete_storage_info_by_index(int n) {
-  if (n >= num_si_)
-    return;
-  const storage_info_message* deleted = get_storage_info_by_index(n);
-  for (int i = n; i < (num_si_ - 1); i++)
-    si_[i] = si_[i+1];
-  num_si_--;
-  si_[num_si_] = nullptr;
-  delete deleted;
-}
+  int certifier::framework::policy_store::get_num_entries() {
+    return num_ents_;
+  }
 
-int certifier::framework::policy_store::get_storage_info_index_by_tag(const string& tag) {
-  for (int i = 0; i < num_si_; i++) {
-    if (si_[i]->tag() == tag)
-      return i;
+  bool certifier::framework::policy_store::add_entry(const string& tag,
+          const string& type, const string& value) {
+    if (num_ents_ >= max_num_ents_)
+      return false;
+    entry_[num_ents_] = new store_entry;
+    entry_[num_ents_]->tag_ = tag;
+    entry_[num_ents_]->type_ = type;
+    entry_[num_ents_]->value_.assign(value.data(), value.size());
+    num_ents_++;
+    return true;
   }
-  return -1;
-}
 
-int certifier::framework::policy_store::get_num_claims() {
-  return num_tc_;
-}
+  int certifier::framework::policy_store::find_entry_by_tag(const string& tag) {
+    for (int i = 0; i < num_ents_; i++) {
+      if (entry_[i]->tag_ == tag)
+        return i;
+    }
+    return -1;
+  }
 
-const claim_message* certifier::framework::policy_store::get_claim_by_index(int n) {
-  if (n >= num_tc_)
-    return nullptr;
-  return &(tc_[n]->claim());
-}
+  int certifier::framework::policy_store::find_entry_by_tag_and_type(const string& tag,
+          const string& type) {
+    for (int i = 0; i < num_ents_; i++) {
+      if (entry_[i]->tag_ == tag && entry_[i]->type_ == type)
+        return i;
+    }
+    return -1;
+  }
 
-bool certifier::framework::policy_store::add_claim(const string& tag, const claim_message& to_add) {
-  if ((num_tc_ + 1) >= max_num_tc_)
-    return false;
-  tagged_claim* t = new(tagged_claim);
-  int n = num_tc_++;
-  t->set_tag(tag);
-  tc_[n] = t;
-  (tc_[n]->mutable_claim())->CopyFrom(to_add);
-  return true;
+void certifier::framework::policy_store::get(int ent, string* v) {
+  if (ent < 0 || ent >= num_ents_)
+    return;
+  *v = entry_[ent]->value_;
 }
 
-void certifier::framework::policy_store::delete_claim_by_index(int n) {
-  if (n >= num_tc_)
+void certifier::framework::policy_store::put(int ent, const string v) {
+  if (ent < 0 || ent >= num_ents_)
```

Comment:
Check for `ent < 0` will not be needed if you changed to `unsigned ent`.

Change interface to `return false` here, if this check fails.